test/controllers/DisassociateDucrControllerSpec.scala

```diff
@@ -16,72 +16,140 @@
 
 package controllers
 
-import base.MovementBaseSpec
-import forms.Choice.AllowedChoiceValues
-import forms.{Choice, DisassociateDucr}
-import org.mockito.ArgumentMatchers._
-import org.mockito.BDDMockito._
+import java.util.UUID
+
+import base.MockFactory.buildSubmissionServiceMock
+import base.testdata.CommonTestData.correctUcr
+import base.{MockAuthConnector, URIHelper}
+import forms.DisassociateDucr
+import org.mockito.ArgumentMatchers.{any, eq => meq}
 import org.mockito.Mockito._
-import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatest.{MustMatchers, WordSpec}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.libs.json.{JsObject, JsString, JsValue}
+import play.api.mvc.Result
+import play.api.test.FakeRequest
 import play.api.test.Helpers._
-import uk.gov.hmrc.http.HeaderCarrier
+import play.filters.csrf.{CSRFConfig, CSRFConfigProvider, CSRFFilter}
+import services.SubmissionService
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.http.SessionKeys
+import utils.FakeRequestCSRFSupport._
+
+import scala.concurrent.Future
+
+class DisassociateDucrControllerSpec
+    extends WordSpec with GuiceOneAppPerSuite with MockAuthConnector with ScalaFutures with MustMatchers
+    with URIHelper {
 
-import scala.concurrent.ExecutionContext
-import scala.concurrent.Future.successful
+  import DisassociateDucrControllerSpec._
 
-class DisassociateDucrControllerSpec extends MovementBaseSpec with BeforeAndAfterEach {
+  implicit val defaultPatience: PatienceConfig =
+    PatienceConfig(timeout = Span(5, Seconds), interval = Span(500, Millis))
 
   private val uri = uriWithContextPath("/disassociate-ducr")
 
-  trait SetUp {
+  private val submissionServiceMock = buildSubmissionServiceMock
```

Comment:
The same with mock here, below you need to add another scenarios. Slightly messy